src/Implementation.fs

```diff
@@ -0,0 +1,95 @@
+module Async_rpc.Implementation
+
+open Core_kernel
+open Async_rpc.Protocol
+open Bin_prot.Common
+open Async_rpc
+open Transport
+
+type 'connection_state t =
+    { rpc_tag: Rpc_tag.t
+      rpc_version: int64
+      partial_fun: 'connection_state
+          -> Bin_prot.Nat0.t Query.t
+          -> buf
+          -> pos_ref
+          -> Transport.Writer.t
+          -> Result.t<unit, Rpc_error.t>
+<<<<<<< HEAD
+=======
+    (*; implement: ('connection_state -> Buffer.t -> Buffer.t)*)
+>>>>>>> ec23e7a927aa331442e8fa83058b083345c54c1d
+     }
+
+let real_function
+    (query_reader: 'query Bin_prot.Type_class.reader)
+    (f: 'connection_state -> 'query -> 'response)
+    response_writer
+    connection_state
+    (query: Bin_prot.Nat0.t Query.t)
+    read_buffer
+    read_buffer_pos_ref
+    transport_writer
+    =
+    Result.let_syntax {
+        let! query_ =
+            let len = query.data
+<<<<<<< HEAD
+=======
+            (*Binary info turned into info f sharp can use*)
+>>>>>>> ec23e7a927aa331442e8fa83058b083345c54c1d
+            Bin_prot_reader.read_and_verify_length
+                query_reader
+                None
+                read_buffer
+                read_buffer_pos_ref
+                len
+                "client-side rpc response un-bin-io'ing"
+<<<<<<< HEAD
+
+=======
+>>>>>>> ec23e7a927aa331442e8fa83058b083345c54c1d
+
+        task {
+            let (resp: Or_error.t<'response>) =
+                Core_kernel.Or_error.try_with (fun () -> f connection_state query_)
+
+            match resp with
+            | Ok resp ->
+                let result: _ Response.t = { id = query.id; data = Ok resp } in
+                let result =
+                    Transport.Writer.send_bin_prot
+                        transport_writer
+                        (Message.bin_writer_needs_length (Writer_with_length.of_writer response_writer))
+                        (Message.t.Response result)
+                    |> Transport.Send_result.to_or_error
+
+                match result with
+                | Ok () -> ()
+                | Error _error ->
```

Comment:
Instead of ignoring all of these results could we print them out so that theres some sort of logging on the server-side? Also we should add a comment about why these are being ignored (since they only fail when there's an issue with the connection in which case the connection will close automatically)